### PR TITLE
Adding Reserves to Hybrid Dispatch

### PR DESCRIPTION
**Pull request by sourabhdalvi**
Some tests are failing because of code moved between PSI and StorageSystemsSimulations. The PR is now ready for review test should be passing.

### CODE REVIEW

**Comment by sourabhdalvi** [comment]
not sure why the Merchant model is failing but are formulation is passing 
image

**Comment by sourabhdalvi** [comment]
@rodrigomha  Hi Can you review the PR

File: src/hybrid_system_constructor.jl [hunk 1]
```diff
@@ -198,6 +198,307 @@ function PSI.construct_device!(
...
+        PSI.ActivePowerBalance,
+        PSI.ActivePowerOutVariable,
+        devices,
+        model,
+        network_model,
+    )
+
+    PSI.add_feedforward_arguments!(container, model, devices)
+
+    if PSI.has_service_model(model)
```
Comment (rodrigomha) [inline src/hybrid_system_constructor.jl:234]: This model is allowed to have services right? I'm not sure how this is working with this error here and then with line 245

File: src/hybrid_system_device_models.jl [hunk 2]
```diff
@@ -855,3 +855,371 @@ function PSI.add_constraints!(
...
+    network_model::PSI.NetworkModel{<:PM.AbstractPowerModel},
+) where {
+    U <: Union{Vector{D}, IS.FlattenIteratorWrapper{D}},
+    V <: PSY.Reserve{PSY.ReserveDown},
+    W <: AbstractHybridFormulation,
+} where {D <: PSY.HybridSystem}
+    time_steps = PSI.get_time_steps(container)
+    resolution = PSI.get_resolution(container)
+    fraction_of_hour = Dates.value(Dates.Minute(resolution)) / PSI.MINUTES_IN_HOUR
+    num_periods =
```
Comment (jd-lara) [inline src/hybrid_system_device_models.jl:876]: Check this is less that 1.

File: src/hybrid_system_device_models.jl [hunk 3]
```diff
@@ -855,3 +855,371 @@ function PSI.add_constraints!(
...
+                    V,
+                    get_name(service),
+                )
+                charge_res_var = PSI.get_variable(
+                    container,
+                    ChargingReserveVariable(),
+                    V,
+                    get_name(service),
+                )
+                service_exp = +discharge_res_var[ci_name, t]
```
Comment (jd-lara) [inline src/hybrid_system_device_models.jl]: use `JuMP.add_to_expression!()` or `PSI.add_to_jump_expression`

**Comment by jd-lara** [comment]
closing this PR, after reviewing the code. This needs a re-write to handle situations like hybrid systems that have heterogenous participation in services.

**Pull request closed without merging**